### PR TITLE
C/CPP: XOR Linked List Documentation

### LINKED ISSUE #6874: C/CPP: XOR Linked List Documentation
## XOR Linked List
A documentation describing XOR Linked List and all its operation in C++

## Directory
Data Structures/Linked List/XOR Linked List

## Description
- What is XOR Linked List?
- Insertion, Deletion, Traversal and Reversal in XOR Linked List
- Advantages and Disadvantages of XOR Linked List

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by OptimalLearner**
<hr>

## Description 📜
Documentation for XOR Linked List Added
Topics covered in the documentation:
- What is XOR Linked List?
- Insertion, Deletion, Traversal and Reversing of XOR Linked List
- C++ Program implementing the above operations on XOR Linked List
- Advantages and Disadvantages of XOR Linked List

Fixes #6874 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [X] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [X] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [X] I have performed a self-review of my own code or work.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] My changes generates no new warnings.
- [X] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by gunjan-g — APPROVED** [review]
@Saviour1001 @Suvraneel LGTM 🚀

**Review by Suvraneel — CHANGES_REQUESTED** [review]
@OptimalLearner 
We're currently not accepting ANY image files separately...
Kindly embed the SVGs within the md itself
(For that just copy image file & paste directly on md)

**Comment by OptimalLearner** [comment]
@Suvraneel
Thank you for your suggestions, I have added the images by copy pasting and removed the separate SVG files.

File: C_CPP/Data Structures/Linked List/XOR Linked List/README.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+previous = current;
+current = next;
+```
+
+# Reversing of XOR Linked List
+Reversing of an XOR linked list is easy as we just need to point our **head** to the **last node** of the linked list.  
+We simply need to traverse the XOR linked list till the end of the linked list and after our current pointer reaches the last node, we update our head to the current node. Now our head is pointing to the last node. When we traverse this linked list, it will be traversed in reverse order.  
+
+# C++ Program to implement XOR Linked List
+```
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Linked List/XOR Linked List/README.md:64]:
```suggestion
```cpp
```

Use syntax Highlighting in all codeblocks

File: C_CPP/Data Structures/Linked List/XOR Linked List/README.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+# Advantages of XOR Linked List
+- **Less** and efficient use of **memory**.
+- **Reversing** an XOR linked list is **easy** as we just need to point the HEAD to the last node.
+
+# Disadvantages of XOR Linked List
+- Little **increase** in code complexity.
+- **Not supported** by many other programming languages like **Java, JavaScript, Python, etc** due to absence of access to memory address of the object.
+- Traversal **not possible** if we need to start it from the **middle** of the XOR linked list as we do not have the **necessary information** to start.
+
+#### Contributor: [Keval Dhanani](https://github.com/OptimalLearner)
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Linked List/XOR Linked List/README.md:217]: @Saviour1001  Allowed?

**Review by Suvraneel — CHANGES_REQUESTED** [review]

File: C_CPP/Data Structures/Linked List/XOR Linked List/README.md (hunk 2 above)
Comment (OptimalLearner) [reply C_CPP/Data Structures/Linked List/XOR Linked List/README.md]: Removed the last line as I noticed no one is using it in their readme files in this domain. I added it as it was allowed in some other domain.

**Comment by Suvraneel** [comment]
@OptimalLearner  LGTM ❤️ Awesome 🚀

**Pull request merged**